src/SDKs/Azure.Base/data-plane/Azure.Base.Tests/Testing/TestLogger.cs

```diff
@@ -22,5 +22,8 @@ public override async Task ProcessAsync(HttpMessage message, ReadOnlyMemory<Http
 
         public override string ToString()
             => _logged.ToString();
+
+        public override void Register(HttpPipelineOptions options)
+            => options.LoggingPolicy = this;
```

Comment:
This doesn't compose well. Registering a new logging policy would remove the old one.

Reply:
Yes, if you don't want it to be replaced, do options.AddPerCallPolicy(this).

Reply:
So what's the recommended way to do it for logging policy implementer? How would the user of API be able to control this behaviour?

Reply:
I was actually thinking about removing the special LoggingPolicy property and always using one of the AddXxxPolicy (configured buy a parameter passed to the ctor of the epolicy:

```c#
if (_perCall) options.AddPerCallPolicy(this);
else options.AddPerRetry(this);
```

But for this, we would need a way to remove the default policies.